consensus/core/src/core.rs

```diff
@@ -2087,14 +2087,15 @@ mod test {
                 D -> [*],
             },
             Round 5 : { 
+                A -> [*],
                 B -> [*],
                 C -> [*],
                 D -> [*],
             },
             Round 6 : { 
-                B -> [A6, B6, C6, D1],
-                C -> [A6, B6, C6, D1],
-                D -> [A6, B6, C6, D1],
+                B -> [A5, B5, C5, D1],
```

Comment:
How did you discover this issue?

Reply:
The linearizer was panicking when calculating the median based timestamp as no ancestors were provided. Even after fixing the test dag to take into account the genesis blocks (part of this PR) , this kept failing. Then realised the issue here.